src/game/puppyprint.c

```diff
@@ -671,17 +732,84 @@ void puppyprint_profiler_process(void) {
         if (sDebugOption >= (sizeof(ppPages) / sizeof(struct PuppyPrintPage))) {
             sDebugOption = 0;
         }
+    } else {
+        if (sPPDebugPage == PUPPYPRINT_PAGE_LEVEL_SELECT)
+        {
+            if (gPlayer1Controller->buttonPressed & U_JPAD) {
+                sLevelSelectOption--;
+                while (sLevelNames[sLevelSelectOption][0] == 0 && sLevelSelectOption < LEVEL_COUNT) {
+                    sLevelSelectOption--;
+                }
+            }
+            if (gPlayer1Controller->buttonPressed & D_JPAD) {
+                sLevelSelectOption++;
+                while (sLevelNames[sLevelSelectOption][0] == 0 && sLevelSelectOption < LEVEL_COUNT) {
+                    sLevelSelectOption++;
+                }
+            }
+            if (sLevelSelectOption == 255) {
+                sLevelSelectOption = (sizeof(sLevelNames)/32)-1;
+                while (sLevelNames[sLevelSelectOption][0] == 0 && sLevelSelectOption < LEVEL_COUNT) {
+                    sLevelSelectOption--;
+                }
+            }
+            if (sLevelSelectOption >= (sizeof(sLevelNames)/32)) {
+                while (sLevelNames[sLevelSelectOption][0] == 0 && sLevelSelectOption < LEVEL_COUNT) {
+                    sLevelSelectOption++;
+                }
+                sLevelSelectOption = 0;
+            }
+            if (gPlayer1Controller->buttonPressed & R_JPAD) {
+                sLevelSelectOptionArea++;
+                if (sLevelSelectOptionArea > 7) {
```

Comment:
```suggestion
                if (sLevelSelectOptionArea > AREA_COUNT - 1) {
```

there may be other instances of this to look out for